File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        lPtr = 0
        profit = 0
        for rPtr in range(1, len(prices)):
            if prices[lPtr] < prices[rPtr]:
                profit = max(profit, prices[rPtr] - prices[lPtr])
            else:
                lPtr = rPtr
        return profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int lPtr = 0;
        
        for (int rPtr = 0; rPtr < prices.length; rPtr++) {
            if (prices[lPtr] < prices[rPtr]) {
                max = Math.max(prices[rPtr] - prices[lPtr], max);
            } else {
                lPtr = rPtr;
            }
        }
        return max;
    }
}